lib/beacon/loader/data_source_module_loader.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Beacon.Loader.DataSourceModuleLoader do
+  @moduledoc false
+  alias Beacon.Loader
+
+  require Logger
+
+  def load_data_source(data, site) do
+    data_source_module = Loader.data_source_module_for_site(site)
+    live_data_functions = Enum.map(data, &live_data_fn/1)
+    # TODO default data
+
+    ast =
+      quote do
+        defmodule unquote(data_source_module) do
+          @behaviour Beacon.DataSource.Behaviour
+          require Logger
+
+          @impl Beacon.DataSource.Behaviour
+          def live_data(path, params) do
+            live_data(path, params, %{})
+          end
+
+          unquote_splicing(live_data_functions)
+
+          def live_data(path, params, data) do
+            Logger.warning("Unhandled Beacon Live Data request for site #{unquote(site)} with path #{inspect(path)} and params #{inspect(params)}")
+            data
+          end
+        end
+      end
+
+    # For debugging - will print module content to the terminal
+    # ast
+    # |> Code.quoted_to_algebra()
+    # |> Inspect.Algebra.format(:infinity)
+    # |> IO.iodata_to_binary()
+    # |> IO.puts()
+
+    :ok = Loader.reload_module!(data_source_module, ast)
+
+    {:ok, data_source_module, ast}
+  end
+
+  defp live_data_fn(live_data) do
+    %{path: path, assigns: assigns} = live_data
+
+    path_list =
+      quote do
+        unquote(
+          path
+          |> String.split("/", trim: true)
+          |> Enum.map(fn
+            ":" <> param -> Macro.var(:"#{param}", :"#{path}")
```

Comment:
Have you considered named parameters (for eg `/v:version`) and glob-like patterns? (for eg `/*page`), ie: support all patterns like `"/pages/he:page/*rest"`?

Reply:
Named params are supported with the Phoenix router syntax i.e. `pages/:page_id/:rest` which gets converted to the "path list" format we used in the previous DataSource module i.e. `["pages", page_id, rest]` where `page_id` and `rest` are usable variables for `:elixir` format LiveDataAssigns

Reply:
@leandrocp I realize now that I misunderstood your question at first.  You are asking about support for additional patterns which would translate to something like:
```elixir
# /foo/bar:baz/bong
["foo", {"bar", baz}, "bong"] = path

# /foo/*rest
# (e.g. /foo/bar/baz/bong)
["foo" | rest] = path
```
I did consider the latter case, at least, but wasn't sure if we wanted to include that in the scope of this PR.  I think it would just require a bit more robust validation of the path, to avoid bad paths such as `foo/*bar/baz`.  Maybe it's good to add that anyway, since this pattern matching here isn't particularly robust (probably breaks on something like `/foo:/bar`).  I could work on this tomorrow, WDYT?